### PR TITLE
Ft accommodation 176153390

### PR DESCRIPTION
**Pull request by Mucyo-Aime-Christian**
## What does this PR do?

- allow the users to interact with accommodations

## Description of Task to be completed?

- create, delete, update, view accommodations
- tests all the accommodation endpoints with mocha
- document endpints with swagger 

## How should this be manually tested?

- clone the Repository 
- use postman or swagger 
- run command ***npm run dev***

## Any background context you want to provide?

- N/A

## What are the relevant pivotal tracker stories?

- https://www.pivotaltracker.com/story/show/176153390

## Screenshots (if appropriate)

- N/A

## Questions:
no questions

### CODE REVIEW

File: src/database/models/user.js [hunk 1]
```diff
@@ -13,11 +13,6 @@ module.exports = (sequelize, DataTypes) => {
 */
     static associate(models) {
       // define association here
-      User.hasMany(models.Trip, {
```
Comment (James-nyagatare) [inline src/database/models/user.js]: @Mucyo-Aime-Christian Why did you delete the associations with trip??

File: src/middlewares/roleCheck.js [hunk 2]
```diff
@@ -0,0 +1,36 @@
+import jwt from 'jsonwebtoken';
```
Comment (James-nyagatare) [inline src/middlewares/roleCheck.js]: @Mucyo-Aime-Christian  I don't think we need this middleware you can use the authentication middleware to check if the user exists and use isAllowed middleware to check if the user is the trip admin.

**Review by James-nyagatare — CHANGES_REQUESTED** [review]
Weldone @Mucyo-Aime-Christian, consider implementing the feedbacks I left tho.

File: src/routes/userRoutes.js [hunk 3]
```diff
@@ -90,6 +90,7 @@ router.post('/signup', signupValidator, UserController.signup);
  *             description: server error.
  * */
 router.post('/login', signinValidator, passportCheck, UserController.signin);
+
```
Comment (harerakalex) [inline src/routes/userRoutes.js:93]: What did you change here, kindly avoid making your PR so noisy, Make sure when you touch the file you are going to make a valid change.

File: src/services/accommodationService.js [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+export default class AccommodationServices {
+  /**
+   * @description get accommodation by name if exists
+   * @param {string} data parameter to be checked against
+   * @return {object} Oject of user if found
+   */
+  static async findAccommodation(data) {
+    try {
+      const accommodation = await Accommodation.findOne({ where: data });
+      return accommodation || null;
```
Comment (harerakalex) [inline src/services/accommodationService.js]: Kindly take time and look into another service to see how others are being implemented.

**Review by harerakalex — CHANGES_REQUESTED** [review]

**Pull request merged**